orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,184 @@
+            if list_options_continue is None:
+                return page_one_response
+
+            if list_options_continue == "next_page":
+                return page_two_response
+
+        with patch('argo.workflows.client.ArchivedWorkflowServiceApi.list_archived_workflows',
+                   side_effect=get_page) as mock_list_archived_workflows:
+            list_results = list(self.client._pull_paginated_results(mock_list_archived_workflows))
+            self.assertEqual(list_results, workflows_on_page_one + workflows_on_page_two)
+            self.assertEqual(mock_list_archived_workflows.call_count, 2)
+
+            mock_list_archived_workflows.assert_has_calls([call(), call(list_options_continue="next_page")])
+
+    def test__pull_paginated_results_api_function_multiple_continues_results_multiple_calls(self):
+        workflows_on_page_one = [
+            mock_argo_workflow('import-hca-total-defg', 'abc123uid', 'Succeeded'),
+            mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+            mock_argo_workflow('import-hca-total-cdef', 'abc345uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset2'
+            }),
+        ]
+
+        page_one_response = self.workflows_to_page_of_results(workflows_on_page_one,
+                                                              "page2")
+
+        workflows_on_page_two = [
+            mock_argo_workflow('import-hca-total-page2', 'page2uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+        ]
+
+        page_two_response = self.workflows_to_page_of_results(workflows_on_page_two,
+                                                              "page3")
+
+        workflows_on_page_three = [
+            mock_argo_workflow('import-hca-1', '12345', 'Succeeded', params={
+                'data-name': 'dataset',
+                'test': 'data'
+            }),
+            mock_argo_workflow('hca-data', 'u123id', 'Succeeded'),
+        ]
+
+        page_three_response = self.workflows_to_page_of_results(workflows_on_page_three,
+                                                                "page4")
+
+        workflows_on_page_four = [
+            mock_argo_workflow('hca-import', 'uid', 'Succeeded', params={
+                'testing_data': 'dataset'
+            }),
+        ]
+
+        page_four_response = self.workflows_to_page_of_results(workflows_on_page_four,
+                                                               None)
+
+        def get_page(list_options_continue=None):
+            {
+                None: page_one_response,
+                "page2": page_two_response,
+                "page3": page_three_response,
+                "page4": page_four_response
+            }[list_options_continue]
+
+            list_results = list(self.client._pull_paginated_results(get_page()))
```

Comment:
You'll need to specify a type for `list_results`, see the most recent failed python validation check.